db/dcrpg/queries.go

```diff
@@ -1290,17 +1333,195 @@ func RetrieveVoutValue(db *sql.DB, txHash string, voutIndex uint32) (value uint6
 	return
 }
 
+// closeRows initiated closing of the Rows after reading all the data
+func closeRows(rows *sql.Rows) {
+	if e := rows.Close(); e != nil {
+		log.Errorf("Close of Query failed: %v", e)
+	}
+}
+
+// RetrieveTicketsPriceByHeight fetches the ticket price and its timestamp that are used
+// to display the ticket price variation on ticket price chart. This data is fetched at an interval
+// of 144 blocks.
+func RetrieveTicketsPriceByHeight(db *sql.DB) (*dbtypes.ChartsData, error) {
+	var items = new(dbtypes.ChartsData)
+	rows, err := db.Query(internal.SelectBlocksTicketsPrice)
+	if err != nil {
+		return nil, err
+	}
+
+	defer closeRows(rows)
+
+	for rows.Next() {
+		var timestamp, price uint64
+		var difficulty float64
+		err = rows.Scan(&price, &timestamp, &difficulty)
+		if err != nil {
+			return nil, err
+		}
+
+		items.Time = append(items.Time, timestamp)
+		items.ValueF = append(items.ValueF, float64(price)/100000000)
```

Comment:
Please use `1e8` or `dcrutil.NewAmount`

Reply:
`dcrutil.NewAmount` seems to accept a float and returns an int64. In the above piece of code I am accepting an uint64 and converting to a float64 value so that I don't have to do the int -> float conversion in the frontend at all. I don't understand why I need to use `dcrutil.NewAmount` here..

Reply:
Right, I read it backwards

Reply:
`dcrutil.Amount(price).ToCoin()`

Reply:
yup, I suggested the same in some edits to the branch: https://github.com/dmigwi/dcrdata/pull/4/files#diff-87f7a577b6af45d0284e4c3fbb98ddf5L1446
To avoid magic numbers and make the semantics obvious, it is worth the import.